lnbits/core/services.py

```diff
@@ -134,26 +133,16 @@ async def pay_invoice(
             # the balance is enough in the next step
             await create_payment(
                 checking_id=temp_id,
-                fee=-fee_reserve(invoice.amount_msat),
+                fee=-fee_reserve_msat,
                 conn=conn,
                 **payment_kwargs,
             )
 
-        # do the balance check if internal payment
-        if internal_checking_id:
-            wallet = await get_wallet(wallet_id, conn=conn)
-            assert wallet
-            if wallet.balance_msat < 0:
-                raise PermissionError("Insufficient balance.")
-
-        # do the balance check if external payment
-        else:
-            if invoice.amount_msat > wallet.balance_msat - (
-                wallet.balance_msat / 100 * 2
-            ):
-                raise PermissionError(
-                    "LNbits requires you keep at least 2% reserve to cover potential routing fees."
-                )
+        # do the balance check
+        wallet = await get_wallet(wallet_id, conn=conn)
+        assert wallet
+        if wallet.balance_msat < 0:
+            raise PermissionError("Insufficient balance.")
```

Comment:
Restore original balance checking, will respect fee for internal vs external payments.